Apps/W1/Shopify/app/src/Invoicing/Codeunits/ShpfyDraftOrdersAPI.Codeunit.al

```diff
@@ -0,0 +1,357 @@
+namespace Microsoft.Integration.Shopify;
+
+using Microsoft.Sales.Comment;
+using Microsoft.Sales.History;
+using Microsoft.Finance.Currency;
+
+/// <summary>
+/// Codeunit Draft Orders API (ID 30159).
+/// </summary>
+codeunit 30159 "Shpfy Draft Orders API"
+{
+    Access = Internal;
+
+    var
+        ShpfyShop: Record "Shpfy Shop";
+        ShpfyCommunicationMgt: Codeunit "Shpfy Communication Mgt.";
+        ShpfyJsonHelper: Codeunit "Shpfy Json Helper";
+
+    /// <summary>
+    /// Creates a draft order in shopify by constructing and sending a graphQL request.
+    /// </summary>
+    /// <param name="TempShpfyOrderHeader">Header information for a shopify order.</param>
+    /// <param name="TempShpfyOrderLine">Line items for a shopify order.</param>
+    /// <param name="ShpfyOrderTaxLines">Tax lines for a shopify order.</param>
+    /// <returns>Unique id of the created draft order in shopify.</returns>
+    internal procedure CreateDraftOrder(
+        var TempShpfyOrderHeader: Record "Shpfy Order Header" temporary;
+        var TempShpfyOrderLine: Record "Shpfy Order Line" temporary;
+        var ShpfyOrderTaxLines: Dictionary of [Text, Decimal]
+    ): BigInteger
+    var
+        GraphQuery: TextBuilder;
+        DraftOrderId: BigInteger;
+    begin
+        GraphQuery := CreateDraftOrderGQLRequest(TempShpfyOrderHeader, TempShpfyOrderLine, ShpfyOrderTaxLines);
+        DraftOrderId := SendDraftOrderGraphQLRequest(GraphQuery);
+        exit(DraftOrderId);
+    end;
+
+    /// <summary>
+    /// Completes a draft order in shopify by converting it to an order.
+    /// </summary>
+    /// <param name="DraftOrderId">Draft order id that needs to be completed.</param>
+    /// <param name="NumberOfLines">Maximum amount of possible fulfillment orders.</param>
```

Comment:
If I fulfill an order line multiple times, do I get multiple fulfillment order? 
This comment could be a bit clearer what do I need to pass in and how it's used.